lib/features/auth/data/repository/location_repository.dart

```diff
@@ -0,0 +1,226 @@
+import 'package:atwoz_app/core/util/util.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:geolocator/geolocator.dart';
+import 'package:geocoding/geocoding.dart';
+
+// 위치 관련 예외 클래스들
+class LocationPermissionDeniedException implements Exception {
```

Comment:
```suggestion
abstract class LocationException implements Exception {
  const LocationException(this.message);
  final String message;
}

class LocationPermissionDeniedException extends LocationException {
  const LocationPermissionDeniedException(): message = '위치 권한이 ...';
}
```

위와 같이 작성하면 message가 수정될 위험도 없고 중복 코드가 줄 것 같습니다~